### PR TITLE
:bug: Fixes #206 Wrong calculation of left and right position of tooltip

### LINKED ISSUE #206: [Bug] getLeft and getRight return wrong result when in middle with a little offset
**Describe the bug**
getLeft and getRight return wrong result when in middle with a little offset.

**Expected behavior**
Text in one line.

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: windows

### PR DESCRIPTION
**Pull request by vatsaltanna**

### CODE REVIEW

**Review by DhavalRKansara — APPROVED** [review]

File: lib/src/tooltip_widget.dart [hunk 1]
```diff
@@ -118,59 +122,43 @@ class _ToolTipWidgetState extends State<ToolTipWidget>
...
+    final descriptionLength = widget.description == null
+        ? 0
+        : (_textSize(widget.description!, descriptionStyle).width +
             widget.contentPadding!.right +
-            widget.contentPadding!.left;
+            widget.contentPadding!.left);
     var maxTextWidth = max(titleLength, descriptionLength);
     if (maxTextWidth > widget.screenSize!.width - 20) {
-      return widget.screenSize!.width - 20;
+      tooltipWidth = widget.screenSize!.width - 20;
```
Comment (DevarshRanpara) [inline lib/src/tooltip_widget.dart:132]: Can we use constant for 20?

File: lib/src/tooltip_widget.dart [hunk 2]
```diff
@@ -118,59 +122,43 @@ class _ToolTipWidgetState extends State<ToolTipWidget>
...
     } else {
-      return maxTextWidth + 15;
+      tooltipWidth = maxTextWidth + 15;
```
Comment (DevarshRanpara) [inline lib/src/tooltip_widget.dart:134]: Can we use constant for 15?

File: lib/src/tooltip_widget.dart [hunk 3]
```diff
@@ -118,59 +122,43 @@ class _ToolTipWidgetState extends State<ToolTipWidget>
...
-      if (leftPadding < 20) {
-        leftPadding = 14;
-      }
-      return leftPadding;
-    } else if (!(_isRight())) {
-      return widget.position!.getCenter() - (_getTooltipWidth() * 0.5);
-    } else {
-      return null;
+    if (widget.position != null) {
+      var leftPositionValue =
```
Comment (DevarshRanpara) [inline lib/src/tooltip_widget.dart:140]: Can we use `double` here instead of `var`?

File: lib/src/tooltip_widget.dart [hunk 4]
```diff
@@ -118,59 +122,43 @@ class _ToolTipWidgetState extends State<ToolTipWidget>
...
+          widget.position!.getCenter() - (tooltipWidth * 0.5);
+
+      return (leftPositionValue + tooltipWidth) >
```
Comment (DevarshRanpara) [inline lib/src/tooltip_widget.dart:143]: Can we remove ternary operators and use conditions to increase readability?

File: lib/src/tooltip_widget.dart [hunk 5]
```diff
@@ -118,59 +122,43 @@ class _ToolTipWidgetState extends State<ToolTipWidget>
...
-          widget.position!.getCenter() + (_getTooltipWidth() / 2);
-      if (rightPadding + _getTooltipWidth() > widget.screenSize!.width) {
-        rightPadding = 14;
-      }
-      return rightPadding;
-    } else if (!(_isLeft())) {
-      return widget.position!.getCenter() - (_getTooltipWidth() * 0.5);
-    } else {
-      return null;
+    if (widget.position != null) {
```
Comment (DevarshRanpara) [inline lib/src/tooltip_widget.dart:154]: Can we remove ternary operators and use conditions to increase readability?

**Review by DevarshRanpara — CHANGES_REQUESTED** [review]

**Review by DevarshRanpara — APPROVED** [review]

**Pull request merged**